src/Polly.Shared/Retry/Alternative.cs

```diff
@@ -0,0 +1,147 @@
+        /// <param name="seed">An optional <see cref="Random"/> seed to use.
+        /// If not specified, will use a shared instance with a random seed, per Microsoft recommendation for maximum randomness.</param>
+        public static IEnumerable<TimeSpan> DecorrelatedJitterBackoff(TimeSpan minDelay, TimeSpan maxDelay, int retryCount, bool fastFirst = false, int? seed = null)
+        {
+            if (minDelay < TimeSpan.Zero) throw new ArgumentOutOfRangeException(nameof(minDelay), minDelay, "should be >= 0ms");
+            if (maxDelay < minDelay) throw new ArgumentOutOfRangeException(nameof(maxDelay), maxDelay, $"should be >= {minDelay}");
+            if (retryCount < 0) throw new ArgumentOutOfRangeException(nameof(retryCount), retryCount, "should be >= 0");
+
+            if (retryCount == 0)
+                yield break;
+
+            var random = new ConcurrentRandom(seed);
+
+            int i = 0;
+            if (fastFirst)
+            {
+                i++;
+                yield return TimeSpan.Zero;
+            }
+
+            double ms = minDelay.TotalMilliseconds;
+            for (; i < retryCount; i++)
+            {
+                // https://github.com/aws-samples/aws-arch-backoff-simulator/blob/master/src/backoff_simulator.py#L45
+                // self.sleep = min(self.cap, random.uniform(self.base, self.sleep * 3))
+
+                // Formula avoids hard clamping (which empirically results in a bad distribution)
+                double ceiling = Math.Min(maxDelay.TotalMilliseconds, ms * 3);
+                ms = random.Uniform(minDelay.TotalMilliseconds, ceiling);
+
+                yield return TimeSpan.FromMilliseconds(ms);
+            }
+        }
+
+        /// <summary>
+        /// Generates sleep durations in an exponential manner.
+        /// The formula used is: Duration = <paramref name="initialDelay"/> x 2^iteration.
+        /// For example: 100ms, 200ms, 400ms, 800ms, ...
+        /// </summary>
+        /// <param name="initialDelay">The duration value for the wait before the first retry.</param>
+        /// <param name="factor">The exponent to multiply each subsequent duration by.</param>
+        /// <param name="retryCount">The maximum number of retries to use, in addition to the original call.</param>
+        /// <param name="fastFirst">Whether the first retry will be immediate or not.</param>
+        public static IEnumerable<TimeSpan> ExponentialBackoff(TimeSpan initialDelay, int retryCount, double factor = 2.0, bool fastFirst = false)
+        {
+            if (initialDelay < TimeSpan.Zero) throw new ArgumentOutOfRangeException(nameof(initialDelay), initialDelay, "should be >= 0ms");
+            if (retryCount < 0) throw new ArgumentOutOfRangeException(nameof(retryCount), retryCount, "should be >= 0");
+            if (factor < 1.0) throw new ArgumentOutOfRangeException(nameof(factor), factor, "should be >= 1.0");
```

Comment:
Not sure about this limits, what it should be. Anything less than `1.0` makes the durations _shrink_ not grow. Perhaps that is a use-case?